src/BlockTypes/ProductQuery.php

```diff
@@ -160,8 +160,10 @@ public function build_query( $query ) {
 			$common_query_values,
 			$this->get_global_query( $parsed_block ),
 			$this->get_custom_orderby_query( $query['orderby'] ),
-			$this->get_queries_by_attributes( $parsed_block ),
-			$this->get_queries_by_applied_filters()
+			$this->get_queries_by_custom_attributes( $parsed_block ),
+			$this->get_queries_by_applied_filters(),
+			$this->get_filter_by_taxonomies_query( $query['tax_query'] ),
```

Comment:
```suggestion
			$this->get_filter_by_taxonomies_query( isset($query['tax_query'] ? $query['tax_query']: array()),
```

Now, if the Products block doesn't have any filter defined, `tax_query` doesn't exist, and WP returns an error.